src/main/java/javaslang/collection/Traversable.java

```diff
@@ -670,6 +722,95 @@ default T last() {
     }
 
     /**
+     * Returns the index of the last occurrence of the given element or -1 if this does not contain the given element.
+     *
+     * @param element an element
+     * @return the index of the last occurrence of the given element
+     */
+    default int lastIndexOf(T element) {
+        return lastIndexOf(element, Integer.MAX_VALUE);
+    }
+
+    /**
+     * Returns the index of the last occurrence of the given element before or at a given end index
+     * or -1 if this does not contain the given element.
+     *
+     * @param element an element
+     * @param end the end index
+     * @return the index of the last occurrence of the given element
+     */
+    int lastIndexOf(T element, int end);
+
+    /**
+     * Finds last index where this sequence contains a given sequence as a slice.
+     * <p>
+     * Note: will not terminate for infinite-sized collections.
+     * @param that the sequence to test
+     * @return the last index such that the elements of this sequence starting a this index match the elements
+     * of sequence that, or -1 of no such subsequence exists.
+     */
+    default int lastIndexOfSlice(Iterable<T> that) {
+        return lastIndexOfSlice(that, Integer.MAX_VALUE);
+    }
+
+    /**
+     * Finds last index before or at a given end index where this sequence contains a given sequence as a slice.
+     * @param that the sequence to test
+     * @param end the end index
+     * @return the last index &lt;= end such that the elements of this sequence starting at this index match
+     * the elements of sequence that, or -1 of no such subsequence exists.
+     */
+    default int lastIndexOfSlice(Iterable<T> that, int end) {
```

Comment:
Also here I would insert `Objects.requireNonNull(that, "that is null");`

and also declare a javadoc param `@throws NullPointerException if that is null`.